tests/unit/lms/services/grouping_test.py

```diff
@@ -84,14 +84,20 @@ def test_if_no_grouping_already_exists_it_inserts_a_new_one(
         assert db_session.query(CanvasGroup).one() == test_grouping
 
     def test_if_a_grouping_already_exists_it_updates_it(
-        self, db_session, upsert_with_parent
+        self, db_session, svc, upsert_with_parent
     ):
+
+        parent = factories.Course(application_instance=svc.application_instance)
```

Comment:
Both need the same parent. 

I'd say this only worked before because both `parent.id`'s was None without a flush.